core/wait/wait_test.go

```diff
@@ -8,66 +8,135 @@ import (
 	"time"
 
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	oapiError "github.com/stackitcloud/stackit-sdk-go/core/oapierror"
 )
 
+// Options used for comparing AsyncActionHandler
+var cmpOpts = []cmp.Option{
+	cmp.AllowUnexported(AsyncActionHandler[interface{}]{}),
+	cmpopts.IgnoreFields(AsyncActionHandler[interface{}]{}, "checkFn"),
```

Comment:
Why are we ingoring the `checkFn` field?

Reply:
Apparently `cmp` can't compare functions

Reply:
Added a comment for this